.travis.yml

```diff
@@ -90,6 +101,26 @@ matrix:
       after_failure:
         - build-support/bin/ci-failure.sh
 
+    - os: linux
+      language: python
+      stage: Deploy Pants Pex
+      env:
+        - PANTS_PEX_RELEASE=stable
+      script:
+        - ./build-support/bin/release.sh -p
+      deploy:
+        # See https://docs.travis-ci.com/user/deployment/releases/
+        provider: releases
+        api_key:
+          secure: "GH+on0frVmTZv0ZE/36ERktsucCkS2Z+KhOqQ/PGUiTOwYm+2D3EWUlaU0lXlIdEzqULXIhCi8ikfmwZMJF+6YxSD8CA+DrGQosgxVQ042OKndxy8IUsaiAezsXp94AGtgrPjIrzNFpUmps9r5hnLCzIK64NjTkZQJIH5LLGupY="
+        file_glob: true
+        file: dist/deploy/pex/*
+        skip_cleanup: true
+        on:
+          # We only release a pex for Pants releases, which are tagged.
```

Comment:
So, while only building pexes for tags should reduce storage costs, Twitter does currently rely on the fact that we travis publishes wheels for "all origin branch shas", in that we consume nightly/master wheels for a nightly smoketest.

I think that the storage costs for pexes should be relatively small, and that most of the cost would be from people actually fetching them. @benjyw : does that sound right?

If so, it would be good to ship pexes for all branches (but not PRs).

Reply:
2 things:
1. This deploys using github releases which only work for tags. I think this is a good thing for our public releases - pypi and github releases are standard discoverable things, an s3 bucket isn't as much.
2. #6016 will require storing pexes for _all_ shas temporarily in S3. This will need a new bucket with a TTL attached to auto-purge. Wheels can get moved there too.

I'd like to defer the per-branch bit until work on point 2.

Reply:
Ok, works for me. Thanks.

As long as the `-p` flag continues to work, Twitter can wait to migrate to the new mechanism.

Reply:
It is true that our costs are 99% bandwidth, not storage.